tests/network/user_defined_network/libudn.py

```
@@ -26,3 +27,22 @@ def lookup_default_pod_ip(pod: Pod) -> str:
+def packet_loss_percent_from_ping_output(ping_output: str) -> float:
+    """Return the packet-loss percentage parsed from ping summary output.
+
+    Args:
+        ping_output: The full output of a ping command.
+
+    Returns:
+        The packet-loss percentage (0-100) reported in the statistics line.
+    """
+    match = re.search(pattern=r"(\d+(?:\.\d+)?)% packet loss", string=ping_output)
+    if not match:
+        raise PacketLossSummaryNotFoundError(f"No packet-loss summary found in ping output: {ping_output}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)